lsp/ty.lua

```diff
@@ -0,0 +1,24 @@
+---@brief
+---
+--- https://github.com/astral-sh/ty
+---
+--- `ty`, a static type checker and language server for python.
+---
+--- The language server can be installed with `uv`:
+--- ```sh
+--- uv tool install ty
+--- ```
+---
+return {
+  cmd = { 'ty', 'server' },
+  filetypes = { 'python' },
+  root_markers = {
+    'ty.toml',
+    'pyproject.toml',
+    'setup.py',
+    'setup.cfg',
+    'requirements.txt',
+    'Pipfile',
+    '.git',
+  },
```

Comment:
Just FYI, ty only looks for `pyproject.toml` and `ty.toml` so we could possibly start with just that and `.git`. We can add other root markers if the need arises.